enumerable.rb

```diff
@@ -0,0 +1,128 @@
+    new_arr
+  end
+
+  def my_all?(*arg)
+    return "`my_all?': wrong # of arguments (given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg[0].is_a? Class
+      my_each { |element| return false unless element.class.ancestors.include?(arg[0]) }
+    elsif arg[0].is_a? Regexp
+      my_each { |element| return false unless arg[0].match(element) }
+    elsif arg.empty?
+      return include?(nil) || include?(false) ? false : true
+    else
+      my_each { |element| return false unless element == arg[0] }
+    end
+    true
+  end
+
+  def my_any?(*arg)
+    return "`my_any?': wrong number of arguments (given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |element| return true if yield(element) }
+    elsif arg.empty?
+      my_each { |element| return true if element }
+      false
+    elsif arg[0].is_a? Class
+      my_each { |element| return true if element.class.ancestors.include?(arg[0]) }
+    elsif arg[0].is_a? Regexp
+      my_each { |element| return true if arg[0].match(element) }
+    else
+      my_each { |element| return true if element == arg[0] }
+    end
+    false
+  end
+
+  def my_none?(*param)
+    return "`my_none': wrong arguments(given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |num| return false if yield(num) }
+    elsif param.empty?
+      my_each { |num| return false unless num.nil? || num == false }
+    elsif param[0].is_a? Regexp
+      my_each { |num| return false if param[0].match(num) }
+    else
+      my_each { |num| return false if num == param[0] }
+    end
+    true
+  end
+
+  def my_count(num = nil)
+    arr = instance_of?(Array) ? self : to_a
+    return arr.length unless block_given? || num
+
+    return arr.my_select { |item| item == num }.length if num
+
+    arr.my_select { |item| yield(item) }.length
+  end
+
+  def my_map(par_ = nil)
+    new_arr = []
+    return to_enum unless block_given?
+
+    if par_
+      my_each { |i| new_arr << par_.call(i) }
+    else
+      my_each { |i| new_arr << yield(i) }
+    end
+    new_arr
+  end
+
+  def my_inject(number = nil, syn = nil)
+    if block_given?
+      acc = number
+      my_each { |i| acc = acc.nil? ? i : yield(acc, i) }
+      raise LocalJumperError unless block_given? || !syn.empty? || !number.empty?
+
+      acc
+    elsif !number.nil? && (number.is_a?(Symbol) || number.is_a?(String))
+      raise LocalJumperError unless block_given? || !number.empty?
+
+      acc = nil
+      my_each { |i| acc = acc.nil? ? i : acc.send(number, i) }
+      acc
+    elsif !syn.nil? && (syn.is_a?(Symbol) || syn.is_a?(String))
+      raise LocalJumperError unless block_given? || !syn.empty?
+
+      acc = number
+      my_each { |i| acc = acc.nil? ? i : acc.send(syn, i) }
+      acc
+    else
+      raise LocalJumperError
+    end
+  end
+end
```

Comment:
- [ ] `my_inject` is not working as the original `inject` method, also is not passing all test cases.

### `inject`` is an Enumerable that has these characteristics:
- Returns a `LocalJumpError` when no block is given and no symbol is passed as an argument.
```
[1, 2, 3].inject
# LocalJumpError (no block given)
```
[Enumerable#inject](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-inject)